cmd/initialize/utils.go

```diff
@@ -12,10 +13,32 @@ import (
 const gitDirPath = ".git"
 const gitignorePath = ".gitignore"
 
-// Check if the .git directory exists
+// Check if we're inside a git repository
 func gitExists() bool {
-	_, err := os.Stat(gitDirPath)
-	return err == nil
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return false
+	}
+
+	_, found := findGitRoot(currentDir)
+	return found
+}
+
+// findGitRoot searches for .git directory up the directory tree
+func findGitRoot(startPath string) (string, bool) {
+	currentPath := startPath
+	for {
+		gitPath := filepath.Join(currentPath, gitDirPath)
+		if _, err := os.Stat(gitPath); err == nil {
+			return currentPath, true
+		}
+
+		parent := filepath.Dir(currentPath)
+		if parent == currentPath {
+			return "", false
+		}
+		currentPath = parent
```

Comment:
What happens if you don't have permissions in the parent directory? Does this still work if you are just in your home dir as a non-sudoer (/home/limitedUser) user that doesn't have permission to read the /home directory or anything above it?

Reply:
The other question I have is just around .gitignore - it would still create it in this directory, right? Which is probably fine, just want to make sure.

Reply:
To the first question, yes, it will work. The only consideration is that if you don't have permission to the .git repository, you will not be able to find the Git root.

To the second question, yes, it will be created in the place where the command was executed.